### DISCUSSION TITLE
ApiGatewayResolver: Avoid having to configure strip_prefixes with pathParameters.proxy

### DISCUSSION THREAD

**Discussion by kbakk** [aws-powertools/powertools-lambda-python#1536] [Q&A]
Hi,

I'm trying to find a way to have the ApiGatewayResolver app be unaware of what the parent resource configured is.

In AWS API Gateway I've configured these resources:

<img width="203" alt="image" src="https://user-images.githubusercontent.com/24817039/191824408-6cfc9269-0b90-4fc5-9216-63e0201ae539.png">

I would like if possible to not have to configure `strip_prefixes` when creating ApiGatewayResolver. Is that an option? 

From the logging I've set up, I see that the message being handled contains a `pathParameters.proxy` - would there be an option to use that when resolving path?

```
  ...
  "message": {
    "body": null,
    "httpMethod": "GET",
    "path": "/control/health/ping",
    "pathParameters": {
      "proxy": "health/ping"
    },
    "requestContext": {
      "path": "/control/health/ping",
      "resourcePath": "/control/{proxy+}",
    ...
```
_Many keys omitted_

Edit: Related discussion (https://github.com/awslabs/aws-lambda-powertools/issues/34) though it doesn't seem like a conclusion on whether to use `pathParameters.proxy` was made.

**Comment by heitorlessa** [comment]
hey @kbakk, I'll come back to you later this week on this. Challenge we had was the discrepancy between REST API Gateway, HTTP API Gateway, and ALB -- when Custom Domain and mapping are used, the payload changes and that `proxy` key wasn't consistent (I need to go back in time to look at payloads again).

The closest to what you're after is using a route prefix: https://awslabs.github.io/aws-lambda-powertools-python/latest/core/event_handler/api_gateway/#route-prefix

**Comment by leandrodamascena** [comment]
I'm closing this discussion as resolved. Please reopen if you need.
